src/goto-checker/bmc_util.cpp

```diff
@@ -225,3 +225,53 @@ void slice(
                    << " remaining after simplification" << messaget::eom;
 }
 
+std::vector<irep_idt> update_properties_status_from_symex_target_equation(
+  propertiest &properties,
+  const symex_target_equationt &equation)
+{
+  std::vector<irep_idt> updated_properties;
+
+  for(const auto &step : equation.SSA_steps)
+  {
+    if(!step.is_assert())
+      continue;
+
+    irep_idt property_id = step.get_property_id();
+
+    if(property_id.empty())
+      continue;
+
+    // Don't set false properties; we wouldn't have traces for them.
+    const auto status = step.cond_expr.is_true() ? property_statust::PASS
+                                                 : property_statust::UNKNOWN;
+    if(properties
+         .emplace(
+           property_id, property_infot{step.source.pc, step.comment, status})
+         .second)
+    {
+      updated_properties.push_back(property_id);
+    }
+    else
+    {
+      property_infot &property_info = properties.at(property_id);
+      property_statust old_status = property_info.status;
+      property_info.status |= status;
+
+      if(property_info.status != old_status)
+        updated_properties.push_back(property_id);
+    }
+  }
+
+  for(auto &property_pair : properties)
+  {
+    if(property_pair.second.status == property_statust::NOT_CHECKED)
```

Comment:
Is there a chance the status was `PASS` before the loop above changed it to `NOT_CHECKED`, and therefore we might spuriously note it as changed? I *think* that can't happen because it's hard to imagine that `|=`result with either `UNKNOWN` or `PASS` on the RHS would turn `PASS` -> `NOT_CHECKED`, but just want to check I've understood.

Reply:
No, cannot happen `|=` is supposed to be monotonic, i.e. once you have a `PASS`, you cannot go back to something undetermined anymore.